src/library/scala/concurrent/Future.scala

```diff
@@ -618,7 +664,7 @@ object Future {
    */
   def sequence[A, M[X] <: TraversableOnce[X]](in: M[Future[A]])(implicit cbf: CanBuildFrom[M[Future[A]], A, M[A]], executor: ExecutionContext): Future[M[A]] = {
     in.foldLeft(successful(cbf(in))) {
-      (fr, fa) => for (r <- fr; a <- fa) yield (r += a)
+      (fr, fa) => fr.zipWith(fa)(_ += _)
```

Comment:
Yep, since 2.12.0-M4 we hoist non-capturing lambda (actually `LambdaMetaFactory` does it for us)

```
scala> val List(f1, f2) = (0 to 1).map(_ => (x: Any) => x).toList
f1: Any => Any = $$Lambda$1329/1454366618@64fe9da7
f2: Any => Any = $$Lambda$1329/1454366618@64fe9da7

scala> f1 eq f2
res0: Boolean = true
```